### PR TITLE
Object.fromEntries: use verifyProperty; add specification details

### PR DESCRIPTION
**Pull request by rwaldron**
https://tc39.github.io/proposal-object-from-entries/

### CODE REVIEW

File: test/built-ins/Object/fromEntries/iterator-closed-for-throwing-entry-value-tostring.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2018 Kevin Gibbons. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-object.fromentries
+description: Closes iterators when toString on a value throws.
```
Comment (bakkot) [inline test/built-ins/Object/fromEntries/iterator-closed-for-throwing-entry-value-tostring.js]: This one isn't valid; only the keys get toString'd (or, to be precise, toPropertyKey'd). The values are passed through without being touched.

File: test/built-ins/Object/fromEntries/iterator-closed-for-null-entry.js [hunk 2]
```diff
@@ -19,7 +36,7 @@ var iterable = {
         advanced = true;
         return {
           done: false,
-          value: 'null',
+          value: null,
```
Comment (bakkot) [inline test/built-ins/Object/fromEntries/iterator-closed-for-null-entry.js]: 👍

File: test/built-ins/Object/fromEntries/iterator-closed-for-throwing-entry-value-accessor.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2018 Kevin Gibbons. All rights reserved.
```
Comment (bakkot) [inline test/built-ins/Object/fromEntries/iterator-closed-for-throwing-entry-value-accessor.js]: Presumably this (and other new tests) should be (C) someone else?

File: test/built-ins/Object/fromEntries/iterator-closed-for-throwing-entry-value-tostring.js (hunk 1 above)
Comment (rwaldron) [reply test/built-ins/Object/fromEntries/iterator-closed-for-throwing-entry-value-tostring.js]: Whoops, yes, that would make sense. I will drop this.

**Comment by rwaldron** [comment]
> Thanks for the followup; sorry for my typos.

You're very welcome and don't worry about the typos :)

File: test/built-ins/Object/fromEntries/iterator-closed-for-throwing-entry-value-accessor.js (hunk 3 above)
Comment (rwaldron) [reply test/built-ins/Object/fromEntries/iterator-closed-for-throwing-entry-value-accessor.js]: Well, I didn't really write a majority of the test, it's your work with only a slight modification

**Pull request merged**